upgrade-guide/upgrade-guides/kubernetes-upgrades/5.8-kubernetes.md

```diff
@@ -0,0 +1,180 @@
+
+{% hint style="success" %}
+You may have changed the name of the `deployment.json` file during your original platform deployment. If so, make sure that you swap up the name wherever it appears in this document.
+{% endhint %}
+
+### Remove components
+
+In the 5.8 templates, the cluster-level components will upgrade to the latest version. You need to remove `kube-prometheus-stack`, `logging-operator app` and `kafka-cluster` from ArgoCD. This change deletes your recent metrics from Grafana and you will only see the latest metrics after you deploy the new `kube-prometheus-stack`. The older CRDs created by `kube-prometheus-stack` and logging-operator charts aren't removed by default during upgrade and should be manually cleaned up with the below commands:
+
+```bash
+kubectl delete crd alertmanagerconfigs.monitoring.coreos.com alertmanagers.monitoring.coreos.com podmonitors.monitoring.coreos.com probes.monitoring.coreos.com prometheusagents.monitoring.coreos.com prometheuses.monitoring.coreos.com prometheusrules.monitoring.coreos.com scrapeconfigs.monitoring.coreos.com servicemonitors.monitoring.coreos.com thanosrulers.monitoring.coreos.com
+```
+
+```bash
+kubectl delete crd clusterflows.logging.banzaicloud.io clusteroutputs.logging.banzaicloud.io flows.logging.banzaicloud.io loggings.logging.banzaicloud.io outputs.logging.banzaicloud.io eventtailers.logging-extensions.banzaicloud.io fluentbitagents.logging.banzaicloud.io hosttailers.logging-extensions.banzaicloud.io nodeagents.logging.banzaicloud.io syslogngclusterflows.logging.banzaicloud.io syslogngclusterflows.logging.banzaicloud.io syslogngclusteroutputs.logging.banzaicloud.io syslogngflows.logging.banzaicloud.io syslogngoutputs.logging.banzaicloud.io
+```
+
+### Upgrade and redeploy components
+
+1.  Open a shell/terminal from the cinchy.devops.automations directory and execute the following command:
+
+    ```bash
+    dotnet Cinchy.DevOps.Automations.dll "deployment.json"
+    ```
+2. Commit all of your changes (if there were any) in each repository.
+3. If there were any changes in your **cinchy.argocd** repository you may need to redeploy ArgoCD.Launch a shell/terminal with the working directory set to the root of the **cinchy.argocd** repository.
+4.  Execute the following command to deploy ArgoCD:
+
+    ```bash
+    bash deploy_argocd.sh
+    ```
+5. Validate ArgoCD pods are running and check that ArgoCD is upgraded v2.7.6 by accessing the ArgoCD application Console.
+6.  Execute the following command to deploy cluster components and Cinchy components:
+
+    ```bash
+    bash deploy_cluster_components.sh
+    bash deploy_cinchy_components.sh
+    ```
+7. You might see a couple of ArgoCD apps out of sync because you deleted the logging operator. Sync them manually. Redis will take a few minutes to recover.
```

Comment:
If you are upgrading from older than 5.7 template version otherwise just sync the argocd apps.